### ISSUE TITLE
hotel (0.4.16) dies randomly (OS X)

### ISSUE THREAD

**Issue by jeanmartin** [typicode/hotel#87]
The hotel daemon stops running at random times. Sometimes right after "hotel start" sometimes later (after hours).

This is what ~/.hotel/daemon.log says:
```
Exiting
Stop daemon
Remove pid file
/usr/local/lib/node_modules/hotel/lib/daemon/app.js:26
  var _req$headers$host$spl = req.headers.host.split(':');
                                              ^

TypeError: Cannot read property 'split' of undefined
    at parseReq (/usr/local/lib/node_modules/hotel/lib/daemon/app.js:26:47)
    at Server.<anonymous> (/usr/local/lib/node_modules/hotel/lib/daemon/app.js:68:21)
    at emitThree (events.js:110:13)
    at Server.emit (events.js:188:7)
    at onParserExecuteCommon (_http_server.js:400:14)
    at HTTPParser.onParserExecute (_http_server.js:368:5)
``` 

I of course did not stop hotel. Running ```hotel start``` after that results in:
```
/Users/me/Library/LaunchAgents/hotel.plist: service already loaded
```

I have to stop and start.

**Comment by typicode** [comment]
Hi @jeanmartin,

Thank you for reporting this issue.

There seem to be a request that doesn't have a host header, and so hotel fails to parse it.
But this code should handle only `.dev` requests, so there should always be a host.

Do you know the request that makes it fail? It would be really helpful if you could add some kind of logs to figure it out, for example:

```
// line 25 in /usr/local/lib/node_modules/hotel/lib/daemon/app.js
console.log(req.headers)
```

Then restart hotel and see what's the output.

**Comment by jeanmartin** [comment]
Done. The last request before seems to be the proxy.pac:
```
21 Apr 14:32:52 - Serve proxy.pac
{ 'proxy-connection': 'keep-alive' }
Exiting
Stop daemon
Remove pid file
/usr/local/lib/node_modules/hotel/lib/daemon/app.js:27
  var _req$headers$host$spl = req.headers.host.split(':');
                                              ^

TypeError: Cannot read property 'split' of undefined
    at parseReq (/usr/local/lib/node_modules/hotel/lib/daemon/app.js:27:47)
    at Server.<anonymous> (/usr/local/lib/node_modules/hotel/lib/daemon/app.js:69:21)
    at emitThree (events.js:110:13)
    at Server.emit (events.js:188:7)
    at onParserExecuteCommon (_http_server.js:400:14)
    at HTTPParser.onParserExecute (_http_server.js:368:5)
```

**Comment by typicode** [comment]
Thanks! What did you do just before it crashed?
By the way, I'll release a fix for this today or this week-end.

**Comment by jeanmartin** [comment]
Nothing. Seriously. I was checking mail, made no request to hotel or my app myself. However you never know if Chrome might have done it (e.g. because network connection was lost and reconnected again or something).
I will not update to your fix immediately and test a bit on the weekend trying to reproduce what exactly happens. Thanks for your fast response!

**Comment by typicode** [comment]
I've released the fix in `v0.4.17`.
No need to spend too much time, except out of curiosity :) let me know if you find out.
In any case, now the daemon checks if there's a `host` header and shouldn't fail anymore.
